docs/design-docs/benchmarks.md

```diff
@@ -0,0 +1,274 @@
+The general process of model training is:
+
+init_distributed_setting -> generate_dataset -> init_dataloader -> create_model -> create_optimizer -> train
+
+The responsibility for function implementation of every layer is as Figure2. These functions will be executed according to the sequence in the figure. The functions that exist in derived class and not in base class are abstract functions.
+
+![Training process and the responsibility for function implementation of every layer](../assets/model-training-process.png)
+
+*Figure 2 Training process and the responsibility for function implementation of every layer*
+
+#### Inference
+
+The general process of the model inference is:
+
+Init_distributed_setting -> generate_dataset -> init_dataloader -> create_model -> inference
+
+Compared with training, it just gets rid of create_optimizer operation. And the responsibility for function implementation of every layer is as Figure 3.
+
+![Inference process and the responsibility for function implementation of every layer](../assets/model-inference-process.png)
+
+*Figure 3 Inference process and the responsibility for function implementation of every layer*
+
+### Micro Benchmarks
+
+The micro-benchmarks have 3-layers Inheritance Relationship. There have two base class for micro-benchmark: 
+One is for pure-python benchmarks named MicroBenchmark. The responsibility for function implementation of every layer is as Figure 4.
+Another is for benchmarks depending on third-party executable program named MicroBenchmarkWithInvoke. The responsibility for function implementation of every layer is as Figure 5.
+
+![micro-benchmarks benchmarking process for MicroBenchmark and the responsibility for function implementation of every layer](../assets/micro-benchmark-process-python.png)
+
+*Figure 4 micro-benchmarks benchmarking process for MicroBenchmark and the responsibility for function implementation of every layer*
+
+![micro-benchmarks benchmarking process for MicroBenchmarkWithInvoke and the responsibility for function implementation of every layer](../assets/micro-benchmark-process-native.png)
+
+*Figure 5 micro-benchmarks benchmarking process for MicroBenchmarkWithInvoke and the responsibility for function implementation of every layer*
+
+### Docker Benchmarks
+
+The Docker benchmarks have 3-layers Inheritance Relationship. And the responsibility for function implementation of every layer is as Figure 6. The DockerBase benchmarks need docker env ready.
+
+![docker-benchmarks benchmarking process and the responsibility for function implementation of every layer](../assets/docker-benchmark-process.png)
+
+*Figure 6 docker-benchmarks benchmarking process and the responsibility for function implementation of every layer*
+
+### Benchmark Registry
+
+Benchmark Registry is designed to
+1.	Provide a way of benchmark registration.
+2.	Avoid modifying existing code when add new benchmarks.
+3.	For models that only have different configs, maximize the reuse of code.
```

Comment:
Reduce the redundant code for model with different configurations.

Reply:
Have changed to:

> Reduce the redundant code for benchmarks with just different configurations.